Lint-Checks.md

```diff
@@ -1,206 +1,188 @@
-* [Pylint Extensions](https://github.com/oppia/oppia/blob/develop/scripts/linters/pylint_extensions.py): Additional pylint checks added as custom extensions.
-* [Linter utils](https://github.com/oppia/oppia/blob/develop/scripts/linters/linter_utils.py): Utility functions shared across all the linters.
+   ```console
+   python -m scripts.linters.pre_commit_linter --only-check-file-extensions py js
+   ```
 
-## How to add new lint checks?
+## Write lint checks
 
-1. Confirm the file type for which you want to add the lint check. 
+If you see a bad pattern cropping up a lot in the code base, you might want to add a lint check to help you find and fix them all. Then, the lint check will ensure no one ever uses that bad pattern again. When adding a new lint check, you should follow the following approach:
 
-2. If it is possible to introduce the test by just adding a rule in the [eslint](https://github.com/oppia/oppia/blob/develop/.eslintrc) (for JS/TS) / [pylint](https://github.com/oppia/oppia/blob/develop/.pylintrc) (for python) / [htmllint](https://github.com/oppia/oppia/blob/develop/.htmllintrc) (for HTML) / [stylelint](https://github.com/oppia/oppia/blob/develop/.stylelintrc) (for CSS) configs, just add the new rule there.
+1. Identify what types of files your lint check needs to evaluate. This is important because we use different linters for different file types.
 
-   * Refer to this documentation for the list of available rules:
+2. Check which of [our linters](#types-of-lint-checks) apply to your file types, and decide which linters you want to use. Add your lint check following the linter-specific instructions below.
 
-      * HTML: https://github.com/htmllint/htmllint/wiki/Options
+3. Run your lint check locally to find all existing violations in the code base. Fix these until the linter passes.
 
-      * CSS: https://github.com/stylelint/stylelint/blob/master/docs/user-guide/get-started.md
+4. Open a PR that both fixes all existing violations and adds the lint check. Once your PR merges, your lint check will prevent any future PRs from introducing new violations.
 
-      * Python: http://pylint.pycqa.org/en/latest/
+## Types of lint checks
 
-      * JS/TS: https://eslint.org/docs/user-guide/configuring
+### Third-party linters
 
-3. If no existing rule is found, add a custom method in the respective linter file for the check to work.
+We make use of several third-party linters, some of which also accept custom rules.
 
-   * If the check is for a python file, prefer adding a custom [pylint extension](https://github.com/oppia/oppia/blob/develop/scripts/linters/pylint_extensions.py) instead of a new method in [python_linter](https://github.com/oppia/oppia/blob/develop/scripts/linters/python_linter.py) file. (Refer this [guide](http://pylint.pycqa.org/en/latest/how_tos/plugins.html) for adding pylint extensions).
+#### ESLint
```

Comment:
Mention that this is run by the JS/TS linter Python file. Ditto for Pylint and html lint runners

Reply:
done